dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -717,10 +719,16 @@ static void AddResponseMessage(ChatCompletionsOptions chatOptions, ChatHistory c
                         streamedContents?.Add(openAIStreamingChatMessageContent);
                         yield return openAIStreamingChatMessageContent;
                     }
+
+                    // Translate all entries into ChatCompletionsFunctionToolCall instances.
+                    toolCalls = OpenAIFunctionToolCall.ConvertToolCallUpdatesToChatCompletionsFunctionToolCalls(
+                        ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
+                    // Translate all entries into FunctionCallContent instances for diagnostics purposes.
+                    functionCallContents = ModelDiagnostics.IsSensitiveEventsEnabled() ? toolCalls.Select(this.GetFunctionCallContent).ToArray() : null;
```

Comment:
The functionCallContents variable should always be initialized by the value returned by GetFunctionCallContent, regardless of whether logging of sensitive events is enabled or not. Otherwise, callers will have to enable sensitive events to have the function call content items in the chat message.  
   
The "if" logic can be moved a few lines below, where the list is passed to the EndStreaming method. Depending on the IsSensitiveEventsEnabled, either the functionCallContents value or null will be passed as an argument.  
   
Since the function call content classes for streaming are not officially supported yet, this behavior may be acceptable for now and can be changed later when adding call content classes for streaming.

Reply:
Agreed. I think we can remove this when we have more consistency across the connectors. I may modify this behavior when I work on the Mistral connector.